### PR TITLE
[5.2] Content Category Save to Menu

### PR DESCRIPTION
**Pull request by brianteeman**
### Summary of Changes
This PR adds two new menu items to the save dropdown menu on the toollbar allowing you to save a categorey directly to a menu item - either as a blog or a list just as you can directly save an article to a menu item.

This is currently restricted to categories in the com_content component. It could be expanded to other components but I didn't see the use case

### Testing Instructions
Check that there are two new items in the save group as shown **only** for categories in com_content and not other places that categories are used such as banners and contacts

image

Check that both new items work as expected and create blog or list menu items as shown for **both** new and existing categories

### Save to Menu as List
image

### Save to Menu as Blog
image

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_categories/src/View/Category/HtmlView.php [hunk 1]
```diff
@@ -203,9 +204,14 @@ protected function addToolbar()
             $saveGroup = $toolbar->dropdownButton('save-group');
 
             $saveGroup->configure(
-                function (Toolbar $childBar) {
+                function (Toolbar $childBar) use ($canDo, $component) {
                     $childBar->save('category.save');
                     $childBar->save2new('category.save2new');
+
+                    if ($canDo->get('core.create', 'com_menus.menu') && $component === "com_content") {
```
Comment (Quy) [inline administrator/components/com_categories/src/View/Category/HtmlView.php]:
```suggestion
                    if ($canDo->get('core.create', 'com_menus.menu') && $component === 'com_content') {
```

File: administrator/components/com_categories/src/View/Category/HtmlView.php [hunk 2]
```diff
@@ -233,6 +239,11 @@ function (Toolbar $childBar) use ($checkedOut, $canDo, $itemEditable) {
                         }
                     }
 
+                    if ($canDo->get('core.create', 'com_menus.menu') && $component === "com_content") {
```
Comment (Quy) [inline administrator/components/com_categories/src/View/Category/HtmlView.php]:
```suggestion
                    if ($canDo->get('core.create', 'com_menus.menu') && $component === 'com_content') {
```

**Comment by rdeutz** [comment]
Is the menu text not a bit imprecise. We have "Save & Close" what does what it says, but the "Save to Menu as List" is more Save & Close & Add List Menu Item. I know it is a bit long but maybe someoen else has a better idea. And yes I know it is the same for article.

**Comment by brianteeman** [comment]
@rdeutz Beyond the scope of this pr for the reasons you stated. If German needs to be - more explicit then it can be done in the translation.

**Comment by PhocaCz** [comment]
I think @rdeutz  point of view is important. It is not so much about the language as about the logic of expression. Moreover, we must realize that English is not only English, but also the base language from which other languages ​​are based.

if I am to understand this feature, then I understand it in the case of: 

**Save & Close & Add List Menu Item**

 but I have no idea what it is about in the case of:
 
 **Save to Menu as List**
 
(and please take this as my personal opinion).

**Comment by rdeutz** [comment]
@LadySolveig I was confused too, so we are two :-)

**Comment by brianteeman** [comment]
It is still out of scope of this PR as it is NOT a new behaviour. In fact Save to menu was added 5 years ago as a result of a survey in the german community to replace an existing toolbar button called new menu item

**Comment by PhocaCz** [comment]
> It is still out of scope of this PR as it is NOT a new behaviour. In fact Save to menu was added 5 years ago as a result of a survey in the german community to replace an existing toolbar button called new menu item

Thanks for the explanation, I understand now, sorry, my ignorance of this feature.

**Comment by exlemor** [comment]
I have tested this item :white_check_mark: successfully on 224214d251eb42b65dcb86e9fcf836bcf97630e0<br /><br />I have tested this successfully.

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43840">issues.joomla.org/tracker/joomla-cms/43840</a>.</sub>

**Comment by brianteeman** [comment]
This has three successful tests - what do I need to do to get the status changed to RTC?

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43840">issues.joomla.org/tracker/joomla-cms/43840</a>.</sub>

**Comment by Hackwar** [comment]
You might just want to give us some time. I do have an eye on this PR. But please also keep in mind that right now if we follow our own rules, we can basically only merge about one PR per hour and I'm not working on Joomla 24/7, so it might take a day or two to process this.

**Comment by brianteeman** [comment]
i was not chasing a merge just RTC so that others didnt waste their time on this one and tested  something else

**Comment by Hackwar** [comment]
Thank you for your contribution @brianteeman! For now this is okay like this, in a future PR I would like to see this not restricted to just com_content, but to extend this to other components as well and make it a bit more flexible.

**Comment by brianteeman** [comment]
> Thank you for your contribution @brianteeman! For now this is okay like this, in a future PR I would like to see this not restricted to just com_content, but to extend this to other components as well and make it a bit more flexible.

Thanks @hackware 

It is not restricted to com_content as such it is just specific to com_content. It can be added to the two other core components that have a category view (newsfeeds and contacts) where it is even simpler as they only have a list view. I can submit a new PR if you wish for those components

**Comment by Hackwar** [comment]
I'd like to have a mechanism which allows also third party extensions which use com_categories to also use this.

**Comment by brianteeman** [comment]
> I'd like to have a mechanism which allows also third party extensions which use com_categories to also use this.

Ah sorry - I misunderstood you. I think that's an easy fix to do - I will look at it once this has been merged

**Pull request merged**